server/services/userService.test.ts

```diff
@@ -23,152 +24,72 @@ const cacheServiceMock = {
-        expect(hmppsAuthClient.getUser).toBeCalledTimes(1)
-        expect(result.roles).toEqual(rolesForMockToken)
-      })
-    })
-
-    describe('with token data', () => {
-      it('Retrieves and formats user name', async () => {
-        const result = await userService.getUser(token, getTokenDataMock())
-        expect(hmppsAuthClient.getUser).toBeCalledTimes(0)
-        expect(result).toEqual({ displayName: 'Token User', name: 'Token User', roles: ['PF_STD_PRISON'] })
-      })
-    })
-
-    it('Propagates error', async () => {
-      hmppsAuthClient.getUser.mockRejectedValue(new Error('some error'))
-
-      await expect(userService.getUser(token)).rejects.toEqual(new Error('some error'))
-    })
-  })
-
-  describe('getUserData', () => {
+  describe('getPrisonUserAccess', () => {
     const prisonApiClient = prisonApiClientMock() as undefined as PrisonApiClient
     beforeEach(() => {
-      hmppsAuthClient = new HmppsAuthClient(null) as jest.Mocked<HmppsAuthClient>
-      hmppsAuthClient.getUser.mockResolvedValue({ ...authUserMock, name: 'john smith' })
-
-      expectedCaseLoads = [{ caseloadFunction: '', caseLoadId: '1', currentlyActive: true, description: '', type: '' }]
       prisonApiClient.getUserCaseLoads = jest.fn(async () => expectedCaseLoads)
       prisonApiClient.getUserLocations = jest.fn(async () => [])
       prisonApiClient.getIsKeyworker = jest.fn(async () => true)
 
-      userService = new UserService(
-        () => hmppsAuthClient,
-        () => prisonApiClient,
-        cacheServiceMock,
-      )
+      userService = new UserService(() => prisonApiClient, cacheServiceMock)
     })
 
     describe('with no cached data', () => {
-      it('Returns case loads', async () => {
-        const { caseLoads } = await userService.getUserData({
-          ...defaultTokenData,
-          authSource: 'nomis',
-          token: TokenMock,
-          user_id: '12345',
-          roles: rolesForMockToken,
-        })
-        expect(prisonApiClient.getUserCaseLoads).toBeCalledTimes(1)
-        expect(caseLoads).toEqual(expectedCaseLoads)
+      beforeEach(() => {
+        cacheServiceMock.getData.mockResolvedValue(null)
       })
 
-      it('Returns active caseload id', async () => {
-        const { activeCaseLoad } = await userService.getUserData({
-          ...defaultTokenData,
-          authSource: 'nomis',
-          token: TokenMock,
-          user_id: '12345',
-          roles: rolesForMockToken,
-        })
+      it('Returns prison user access data', async () => {
+        const userAccess = await userService.getPrisonUserAccess(prisonUserMock)
+
         expect(prisonApiClient.getUserCaseLoads).toBeCalledTimes(1)
-        expect(activeCaseLoad).toEqual(expectedCaseLoads[0])
+        expect(prisonApiClient.getUserLocations).toBeCalledTimes(1)
+        expect(prisonApiClient.getIsKeyworker).toBeCalledTimes(1)
+
+        expect(userAccess).toEqual(expectedUserAccess)
```

Comment:
Collapsing tests for individual parts of the object down into one check for the `expectedUserAccess`